### PR TITLE
Rust 2018 Phase 2

### PR DESCRIPTION
**Pull request by kkdawkins**
Upgrade to 2018 for:

- edgelet-http-workload
- edgelet-iothub
- edgelet-test-utils
- edgelet-utils
- hsm-rs
- hsm-sys
- hyper-named-pipe
- iotedge
- iotedged

### CODE REVIEW

File: edgelet/edgelet-test-utils/src/lib.rs [hunk 1]
```diff
@@ -9,29 +9,8 @@
     clippy::use_self
 )]
 
-extern crate chrono;
-extern crate edgelet_core;
 #[macro_use]
```
Comment (darobs) [inline edgelet/edgelet-test-utils/src/lib.rs]: Was this intentionally left behind?

File: edgelet/edgelet-utils/src/lib.rs [hunk 2]
```diff
@@ -4,38 +4,17 @@
 #![deny(clippy::all, clippy::pedantic)]
```
Comment (darobs) [inline edgelet/edgelet-utils/src/lib.rs]: Missed a "unused_extern_crates"?

File: edgelet/edgelet-test-utils/src/lib.rs (hunk 1 above)
Comment (kkdawkins) [reply edgelet/edgelet-test-utils/src/lib.rs]: That's really weird that it built with those in place...

Comment (kkdawkins) [reply edgelet/edgelet-test-utils/src/lib.rs]: Good catch!

**Pull request merged**